### ISSUE TITLE
Out of memory crash when deleting an address via the address repository

### ISSUE THREAD

**Issue by cherreman** [magento/magento2#8351]
### Preconditions

1. Magento 2.1.2
2. PHP 7.0.15

### Steps to reproduce

1. Create a customer with a lot of addresses (+200 in our case)
2. Delete one of the addresses via the AddressRepository

### Expected result

1. The address is deleted

### Actual result

1. Out of memory crash

In the delete() and deleteById() methods in AddressRepository, the addresses of the customer get cleared. For this, it first retrieves all addresses of the customer. Assuming the bidirectional relation between customer and address potentially uses a large amount of memory, this will eventually lead to a crash.

Would it be feasible to remove the reference from address to customer? If we see customer as an aggregate root, then it can have address objects, though the backreference from address to customer should probably only be the customer ID.

```
public function deleteById($addressId)
    {
        $address = $this->addressRegistry->retrieve($addressId);
        $customerModel = $this->customerRegistry->retrieve($address->getCustomerId());
        $customerModel->getAddressCollection()->clear();
        $this->addressResource->delete($address);
        $this->addressRegistry->remove($addressId);
        return true;
    }
```

**Comment by dnadle** [comment]
Still an issue in 2.1.6.

**Comment by magento-engcom-team** [comment]
@cherreman, thank you for your report.
The issue is already fixed in develop branch, 2.2.0
